src/awkward/operations/ak_mean.py

```diff
@@ -225,8 +225,8 @@ def _impl(x, weight, axis, keepdims, mask_identity, highlevel, behavior, attrs):
             sumw = ak.operations.ak_sum._impl(
                 x * 0 + weight,
                 axis,
-                keepdims,
-                mask_identity,
+                keepdims=True,
+                mask_identity=True,
```

Comment:
This wasn't in my original issue, but was brought up by @ianna in the discussion, so I decided to fix it as well; added tests for it as well